src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol.cs

```diff
@@ -1118,6 +1118,17 @@ protected sealed override void DecodeWellKnownAttributeImpl(ref DecodeWellKnownA
                 // CS1608: The Required attribute is not permitted on C# types
                 diagnostics.Add(ErrorCode.ERR_CantUseRequiredAttribute, arguments.AttributeSyntaxOpt.Name.Location);
             }
+            else if (attribute.IsTargetAttribute(this, AttributeDescription.InlineArrayAttribute))
+            {
+                if (arguments.GetOrCreateData<TypeWellKnownAttributeData>().InlineArrayLength == 0)
```

Comment:
> Why don't we use ```arguments.HasDecodedData``` to detect that attribute hasn't been decoded yet?

Because that property doesn't return information specific to this attribute.

Reply:
I guess this is fine, but seems that using `HasDecodedData` here would be useful. We would not need two special values (0 and -1) for `InlineArrayLength` (and mapping between them below) - the `if` would just check for `!HasDecodedData`. We could also have `VerifySealed(expected: true);` in getter of `TypeWellKnownAttributeData.InlineArrayLength`.

Reply:
> I guess this is fine, but seems that using `HasDecodedData` here would be useful. We would not need two special values (0 and -1) for `InlineArrayLength` (and mapping between them below) - the `if` would just check for `!HasDecodedData`. We could also have `VerifySealed(expected: true);` in getter of `TypeWellKnownAttributeData.InlineArrayLength`.

I guess I am missing how it could help. The ```HasDecodedData``` property simply tells us if an instance of ```TypeWellKnownAttributeData``` has been allocated for this type. It doesn't tell us why and what data are actually stored in in that instance. If the property returns true, that tells us nothing about whether we have seen any specific well known attribute before.

What am I missing?